src/libostree/ostree-sepolicy.c

```diff
@@ -155,6 +159,92 @@ ostree_sepolicy_class_init (OstreeSePolicyClass *klass)
                                                         G_PARAM_READWRITE | G_PARAM_CONSTRUCT_ONLY));
 }
 
+#ifdef HAVE_SELINUX
+
+/* Find the latest policy file in our root and return its checksum. */
+static gboolean
+get_policy_checksum (char **out_csum,
+                     GCancellable *cancellable,
+                     GError **error)
+{
+  gboolean ret = FALSE;
+  const char *binary_policy_path = selinux_binary_policy_path ();
+  g_autofree char *bindir_path = g_path_get_dirname (binary_policy_path);
+  g_autoptr(GFile) bindir = g_file_new_for_path (bindir_path);
+  g_autoptr(GFile) best_child = NULL;
+  int best_version = 0;
+
+  /* The binary_policy_path includes the prefix of the basename, but without the
+   * period, e.g. "policy". */
+  g_autofree char *binfile_prefix = g_path_get_basename (binary_policy_path);
+  g_autofree char *binfile_no_fnmatch = g_strdup_printf ("%s.*[!0-9]*",
+                                                         binfile_prefix);
+
+  g_autoptr(GFileEnumerator) direnum =
+    g_file_enumerate_children (bindir, OSTREE_GIO_FAST_QUERYINFO,
+                               G_FILE_QUERY_INFO_NOFOLLOW_SYMLINKS,
+                               cancellable, error);
+
+  if (!direnum)
+    goto out;
+
+  while (TRUE)
+    {
+      GFile *child;
+      GFileInfo *fi;
+      GFileType ftype;
+
+      if (!gs_file_enumerator_iterate (direnum, &fi, &child, cancellable, error))
+        goto out;
+      if (!fi)
+        break;
+
+      ftype = g_file_info_get_file_type (fi);
+      if (ftype == G_FILE_TYPE_REGULAR)
+        {
+          g_autofree char *name = g_file_get_basename (child);
+
+          /* poor man's regex */
+          if (g_str_has_prefix (name, binfile_prefix) &&
+              fnmatch (binfile_no_fnmatch, name, FNM_NOESCAPE))
```

Comment:
It confused me initially that `fnmatch` returns `0` if it matched.  Can we do `fnmatch (...) == FNM_NOMATCH` ?